03_letters/patrickmarabeas/main.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+// Letter frequency map
+func letters(s string) map[rune]int {
+	result := make(map[rune]int)
+	// Increment key's value by one for each instance
+	// Starts with the zero value of the ValueType (int)
+	for _, v := range s {
+		result[v]++
+	}
+
+	return result
+}
+
+// Sorted letter kvp slice
+func sortLetters(m map[rune]int) []string {
+	length := len(m)
+	result := make([]string, length)
+	// Create a slice of runes from map keys
+	runes := make([]rune, length)
+	i := 0
+	for k := range m {
+		runes[i] = k
+		i++
+	}
+	// Sort the slice of runes
+	// The less function that sort implements takes int32 which aligns with rune
```

Comment:
I believe these comments are unnecessary, the code should be self explanatory

Reply:
As a Go scrub, it really isn't. Initially I was using casting to solve this - but that felt wrong. This documents *for myself* how I got around the need to do so, and why this works.